packages/edit-widgets/src/components/widget-areas-block-editor-provider/index.js

```diff
@@ -55,6 +51,11 @@ export default function WidgetAreasBlockEditorProvider( {
 			? getEntityRecord( 'root', 'site' )
 			: undefined;
 		return {
+			hasUploadPermissions:
+				canUser( 'create', {
+					kind: 'root',
+					name: 'media',
+				} ) ?? true,
```

Comment:
I simplified the check. It matches how the `Customize Widgets` and `Editor` packages are doing the same check.

https://github.com/WordPress/gutenberg/blob/cdf31e8677db66c021aaae0b02d759b99a05328b/packages/customize-widgets/src/components/sidebar-block-editor/index.js#L53-L57